src/DynamoRevit/DynamoRevit.cs

```diff
@@ -890,4 +912,85 @@ public override IEnumerable<string> GetDynamoUserDataLocations()
             return paths;
         }
     }
+
+    [Serializable()]
+    public class ExtensionInfo
+    {
+        public string NodePath { get; set; }
+        public string LayoutSpecsPath { get; set; }
+    }
+    internal static class DynamoRevitInternalNodes
+    {
+        private static IEnumerable<string> GetAllExtensionFiles()
+        {
+            string currentAssemblyPath = Assembly.GetExecutingAssembly().Location;
+            string currentAssemblyDir = Path.GetDirectoryName(currentAssemblyPath);
+
+            string internalNodesDir = Path.Combine(currentAssemblyDir, "packages");
+            if (false == Directory.Exists(internalNodesDir))
+            {
+                return new List<string>();
+            }
+
+            string[] internalNodesFolders = Directory.GetDirectories(internalNodesDir);
+
+            List<string> extensionFiles = new List<string>();
+            foreach (string dir in internalNodesFolders)
+            {
+                string extensionFile = Path.Combine(dir, "extension.xml");
+                if (true == File.Exists(extensionFile))
+                {
+                    extensionFiles.Add(extensionFile);
+                }
+            }
+            return extensionFiles;
+        }
+        private static IEnumerable<ExtensionInfo> ParseExtensionFiles(IEnumerable<string> extensionFiles)
+        {
+            List<ExtensionInfo> extensionInfos = new List<ExtensionInfo>();
+
+            foreach (string extensionFile in extensionFiles)
+            {
+                try
+                {
+                    string extensionFileDir = Path.GetDirectoryName(extensionFile);
+                    using (StreamReader reader = new StreamReader(extensionFile))
+                    {
+                        XmlSerializer serializer = new XmlSerializer(typeof(ExtensionInfo));
+                        ExtensionInfo extInfo = serializer.Deserialize(reader) as ExtensionInfo;
+
+                        // convert to absolute path, if needed
+                        if (false == Path.IsPathRooted(extInfo.NodePath))
+                        {
+                            extInfo.NodePath = Path.Combine(extensionFileDir, extInfo.NodePath);
+                        }
+
+                        // convert to absolute path, if needed
+                        if (false == Path.IsPathRooted(extInfo.LayoutSpecsPath))
+                        {
+                            extInfo.LayoutSpecsPath = Path.Combine(extensionFileDir, extInfo.LayoutSpecsPath);
+                        }
+
+                        extensionInfos.Add(extInfo);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(string.Format("Exception while trying to parse extension file {0}", extensionFile));
```

Comment:
if this will be shown to users it should live in a .resx - if it's just for internal devs, than this should be fine.

Reply:
this is only for internal devs